src/components/Autocomplete/types.ts

```diff
@@ -3,7 +3,7 @@ import { AutocompleteProps } from '@material-ui/lab';
 import { WithClassAndId, WithCssGeneratorFunction } from '../../core';
 import { LabelWithProps } from '../shared';
 
-export type OptionType = { label: string; value: string };
+export type OptionType = { label: string; category?: React.ReactNode | string; value: string };
```

Comment:
Alicja needs this to be exported and for it to be flexible. You're not exporting this beyond your component and you're just adding more props to this... This `category` you added doesn't make any sense... Is it something that platform needs? If so do you think it belongs here? What Alicja needs from you is a flexible type and this isn't one...

```suggestion
export type OptionType = { label: string; value: string; [key: string]: any };
```